### DISCUSSION TITLE
Extension - TCN

### DISCUSSION THREAD

**Discussion by d11pshields** [supabase/supabase#5570] [Questions]
It looks like Supabase supports TCN.  Is there an example of TCN working in Supabase?

Also, should **pg_notify()** work in Supabase?

**Comment by steve-chavez** [comment]
> Also, should pg_notify() work in Supabase?

No example of [tcn](https://www.postgresql.org/docs/current/tcn.html), but yes, NOTIFY should work. You'll need a backend component that connects to the db and listens on the `tcn` channel(`LISTEN tcn`).

Note that NOTIFY has a payload limit of 8KB, so perhaps you'd be better with [realtime](https://github.com/supabase/realtime/#why-not-just-use-postgresqls-notify).

**Reply by d11pshields** [reply]
Thanks Steve.

I've tried NOTIFY with non-Supabase Postgres databases and it works fine.  But it doesn't seem to work with Supabase?  I have tried enabling TCN on my Supabase database, but this seems to have no effect.

**Reply by steve-chavez** [reply]
> I've tried NOTIFY with non-Supabase Postgres databases and it works fine. But it doesn't seem to work with Supabase?

Do you have a minimal example of what you're trying?

**Reply by d11pshields** [reply]
Thanks Steve, I really appreciate your help.

I've got a background service using **LISTEN datachange** and an event should be generated when a notification is received.   

To simplify testing, I've then been testing direct from Postgres using **NOTIFY datachange**

I've tried this approach with non-Supabase databases and it works fine and an event is generated within my background service.  I'm also able to notify with payloads etc.    But with Supabase my background service does not receive any notifications?

Do you think you might be able to test a simple LISTEN tcn / NOTIFY tcn test at your end just to make sure LISTEN/NOTIFY works ok in Supabase?  Or failing that, provide some example code for LISTEN/NOTIFY?

My background service and approach is essentially based around something similar to this:
[https://www.graymatterdeveloper.com/2019/12/02/listening-events-postgresql/](url)

**Reply by steve-chavez** [reply]
The simplest test I've done is to connect to `psql` in one terminal, `LISTEN`, and then from another terminal do a `NOTIFY` - that seems to be working good:

![2022-02-22-100133_979x301_scrot](https://user-images.githubusercontent.com/1829294/155159716-35b6ebfd-5fab-4c90-8485-6709799ad7db.png)

Can you try that and see if it works?

**Reply by d11pshields** [reply]
Thanks Steve, I followed your example and was able to successfully generate and receive notifications using LISTEN in one terminal window and NOTIFY in another terminal window.

Therefore, the issue must be at my end.  I will do some further testing...

Many thanks for your help.

**Reply by d11pshields** [reply]
Steve, I found the issue.  The reason why the PSQL test worked was that LISTEN/NOTIFY uses the default port.   I have been using a different port number, as specified in the Supabase "Connection Info" panel.

Thanks for your help.

**Reply by steve-chavez** [reply]
> I have been using a different port number, as specified in the Supabase "Connection Info" panel.

@d11pshields Good to know you were able to solve your issue. Which was the port you were using?

**Reply by saravananravi08** [reply]
port -> 5432 ,  faced the same scenario and adding it for further reference
